docs/reference/esql/esql-index-options.asciidoc

```diff
@@ -0,0 +1,42 @@
+[[esql-index-options]]
+=== {esql} index options
```

Comment:
> they kinda belong together.

I initially added this option right after `FROM`. But this section is a bit larger than a simple parameter explanation, so I wanted to pull it out. And I didn't want to combine it with METADATA, since IMO they _are_ different topics (what extra fields to pull in vs. how to resolve indices), even if provided to the same `FROM` command.

> The 'Learning ES|QL' part of our docs is beginning to look a bit crowded

It's just that there's (growing amount of) stuff to be communicated. IMO we might revisit the "Learning" bit of the section name (as we're quite exhaustive in it).

Reply:
> IMO we might revisit the "Learning" bit of the section name (as we're quite exhaustive in it).

@bpintea might indeed be better to call this the _ES|QL reference_

Reply:
@leemthompo, we now also have `Syntax reference` (that could then maybe become just `Syntax`?), but otherwise this proposal would LGTM.